src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -0,0 +1,403 @@
+                " save method for each model separately with a different path."
+            )
+        rank = self.local_rank
+        world_size = self.world_size
+        import torch_xla.core.xla_model as xm
+
+        # ensure model parameters are updated
+        xm.mark_step()
+
+        converted_state: Dict[str, Any] = {}
+        for key, obj in state.items():
+            # convert the state
+            if isinstance(obj, Module) and isinstance(obj, XLAFSDP):
+                converted = obj.state_dict()
+                # add shard_metadata to state
+                converted_state["shard_metadata"] = obj.get_shard_metadata()
+            elif isinstance(obj, Optimizer):
+                converted = obj.state_dict()
+            else:
+                converted = obj
+            _apply_filter(key, filter or {}, converted, converted_state)
+
+        self.checkpoint_io.save_checkpoint(
+            converted_state,
+            os.path.join(path, f"checkpoint_rank-{rank:08d}-of-{world_size:08d}.pth"),
+            storage_options=storage_options,
+        )
+
+        if self._state_dict_type == "full":
+            from torch_xla.distributed.fsdp import consolidate_sharded_model_checkpoints
+
+            if self.is_global_zero:
+                consolidate_sharded_model_checkpoints(
+                    ckpt_prefix=os.path.join(path, "checkpoint"),
+                    ckpt_suffix="_rank-*-of-*.pth",
+                )
+            self.barrier("ckpt_consolidation")
+            self.checkpoint_io.remove_checkpoint(
+                os.path.join(path, f"checkpoint_rank-{rank:08d}-of-{world_size:08d}.pth")
+            )
+
+    def remove_checkpoint(self, folderpath: _PATH) -> None:
+        """Remove checkpoint filepath from the filesystem.
+
+        Args:
+            filepath: Path to folder with full/sharded checkpoint(s)
+        """
+        rank = self.local_rank
+        world_size = self.world_size
+
+        # broadcast the path from rank 0 to ensure all the states are loaded from a common path
+        folderpath = self.broadcast(folderpath)
+        if not os.path.isdir(folderpath):
+            raise NotImplementedError(
+                "The `XLAFSDPStrategy` requires specifying the directory where to remove checkpoints."
+            )
+        if self._state_dict_type == "sharded":
+            self.checkpoint_io.remove_checkpoint(
+                os.path.join(folderpath, f"checkpoint_rank-{rank:08d}-of-{world_size:08d}.pth")
+            )
+        elif self._state_dict_type == "full":
+            self.checkpoint_io.remove_checkpoint(os.path.join(folderpath, "checkpoint_consolidated.pth"))
+        else:
+            raise ValueError(f"Unknown state_dict_type: {self._state_dict_type}")
+
+    def load_checkpoint(
+        self,
+        path: _PATH,
+        state: Optional[Union[Module, Optimizer, Dict[str, Union[Module, Optimizer, Any]]]] = None,
```

Comment:
FSDP recently added support for loading only one module or optimizer object. I don't think it's important but we should at least throw a `NotImplementedError` in that case

https://github.com/Lightning-AI/lightning/blob/master/src/lightning/fabric/strategies/fsdp.py#L510-L517

Reply:
will add